### PR TITLE
Changed DataFilters to have overflow

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataFilters to have overflow when using `drop` or `layer`.

The snapshot changes are primarily due to two reasons:
- the restructuring of DataForm
- adding `flex={false}` to FormField

#### Where should the reviewer start?

DataForm.js

#### What testing has been done on this PR?

storys

#### How should this be manually tested?

storys

#### Do Jest tests follow these best practices?

no tests were changed in the creation of this pull request

#### What are the relevant issues?

#6626 

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataFilters/DataFilters.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ export const DataFilters = ({ drop, children, heading, layer, ...rest }) => {
   const content = (
     <DataForm
       pad={controlled ? 'medium' : undefined}
-      gap="small"
+      gap="xsmall"
```
Comment (taysea) [inline src/js/components/DataFilters/DataFilters.js:89]: HPE FormFields already have spacing built into FormField styling, so `gap` shouldn't be used at all. Wondering if this value should be removed or theme driven?

File: src/js/components/DataFilters/DataFilters.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ export const DataFilters = ({ drop, children, heading, layer, ...rest }) => {
               }))
           : undefined
       }
-      {...(!controlled ? rest : {})}
+      {...(!controlled ? rest : { fill: 'vertical' })}
```
Comment (taysea) [inline src/js/components/DataFilters/DataFilters.js:103]: Is controlled when the user isn't passing their own DataFilter? it feels like fill: vertical would always be helpful if that enables the overflow scroll.

File: src/js/components/DataFilters/DataFilters.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ export const DataFilters = ({ drop, children, heading, layer, ...rest }) => {
   );
 
   const clearControl = badge && (
-    <Box flex={false}>
+    <Box flex={false} margin={{ start: 'small' }}>
```
Comment (taysea) [inline src/js/components/DataFilters/DataFilters.js:52]: Not sure where to comment on it, but the gap between the layer footer buttons is too large. In HPE patterns, we use gap="small"

File: src/js/components/DataFilters/DataFilters.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/DataFilters/DataFilters.js:103]: When rendering inline, it allows the caller to decide how to handle it. Do you have a specific scenario you are envisioning?

File: src/js/components/DataFilters/DataFilters.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataFilters/DataFilters.js:89]: Good catch. I've made a few layout adjustments to address.

File: src/js/components/DataFilters/DataFilters.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/DataFilters/DataFilters.js:52]: FYI, the default `gap` in Footer is "medium". Explicitly made it "small" for DataForm.

**Review by taysea — APPROVED** [review]
LGTM!

**Pull request merged**